src/App.js

```diff
@@ -1,35 +1,24 @@
-import { DataQuery } from '@dhis2/app-runtime'
-import i18n from '@dhis2/d2-i18n'
+import { CssVariables } from '@dhis2/ui'
 import React from 'react'
-import classes from './App.module.css'
+import './App.css'
+import { Layout, Header, Main, Sidebar, Footer } from './layout/index.js'
 
-const query = {
-    me: {
-        resource: 'me',
-    },
-}
-
-const MyApp = () => (
-    <div className={classes.container}>
-        <DataQuery query={query}>
-            {({ error, loading, data }) => {
-                if (error) {
-                    return <span>ERROR</span>
-                }
-                if (loading) {
-                    return <span>...</span>
-                }
-                return (
-                    <>
-                        <h1>
-                            {i18n.t('Hello {{name}}', { name: data.me.name })}
-                        </h1>
-                        <h3>{i18n.t('Welcome to DHIS2!')}</h3>
-                    </>
-                )
-            }}
-        </DataQuery>
-    </div>
+const App = () => (
+    <Layout>
+        <CssVariables colors />
+        <Header>
+            <h1>Header</h1>
+        </Header>
+        <Main>
+            <h1>Main</h1>
+        </Main>
+        <Sidebar>
+            <h1>Sidebar</h1>
+        </Sidebar>
+        <Footer>
+            <h1>Footer</h1>
+        </Footer>
```

Comment:
I thought about this a bit. I think, until we have sub-grids, it makes more sense to put the bottom bar into the workspace area. On the one hand, the bottom bar is functionally tightly coupled to the workspace (it's basically the submit buttons), on the other hand is the "style behavior" (the animation when the right-hand panel opens/closes) identical/simultaneously, applying that effect on one container only seems reasonable to me as it's slightly more maintainable and easier to reason about.

Reply:
I've just finished the layout component. So with the current layout I've gone for an order in the html that makes sense for the html, and I'm just using grid for positioning/animating. That's also what I've always seen recommended as you don't have to care about source order with grid. So I don't think we have to overthink this, the simplest order seems to me to be just header, content, aside, footer (all on the same level). Then we can just lay that out in css however we want.